### ISSUE TITLE
Wait on undecided fulfillments

### ISSUE THREAD

**Issue by rhsimplex** [bigchaindb/bigchaindb#670]
From @r-marques comments in #629:

> There may be some logic that we may need to handle.
> 
> For the case in which an input is in an undecided block. This does not necessarily mean that the transaction is invalid, it just needs to wait a bit more for block in which the input resides to be validated.
> 
> The solutions would be:
> 1. Leave this as it is and make it the clients responsibility to check that the inputs were correctly validated before using them.
> 2. Add additional logic in the block creation to cache transactions in which inputs are still in an undecided state and try again later.
> 
> If we feel that point 2. is worth it can create an issue for it and do it in a separate pull request as an optimization.

**Comment by r-marques** [comment]
The server is following option 1. All inputs of a transaction need to be valid.
